### PR TITLE
WIP:  #689 - Single flow for docker and non-docker way of running script

### PR DESCRIPTION
**Pull request by grzesuav**

### CODE REVIEW

**Comment by karianna** [comment]
See #689

**Comment by grzesuav** [comment]
@karianna currently we configure build on host computer, then copy config file inside docker and use it there. It ends up with misconfigured build which fails.

The main reason of this change is to have consistent build. So idea was to extract docker part to separate script. Build should behave the same, ran native or in docker.

Does it makes more sense ?

File: docker/jdk8/x86_64/ubuntu/Dockerfile [hunk 1]
```diff
@@ -54,7 +54,7 @@ RUN apt-get update \
 RUN mkdir -p /openjdk/target
 
 COPY sbin /openjdk/sbin
-COPY workspace/config /openjdk/config
+COPY *.sh /openjdk/
```
Comment (karianna) [inline docker/jdk8/x86_64/ubuntu/Dockerfile]: Prefer explicit list of files than wildcard (security risk)

File: makejdk-any-platform-docker.sh [hunk 2]
```diff
@@ -146,10 +147,38 @@ buildOpenJDKViaDocker()
        "${gitSshAccess[@]}" \
        -e DEBUG_DOCKER_FLAG="${BUILD_CONFIG[DEBUG_DOCKER]}" \
        -e BUILD_VARIANT="${BUILD_CONFIG[BUILD_VARIANT]}" \
-       "${dockerEntrypoint[@]}"
-  
+       "${dockerEntrypoint[@]}" \
+        "${BUILD_CONFIG[CONTAINER_NAME]}" "$@"
```
Comment (karianna) [inline makejdk-any-platform-docker.sh]: whitespace (indented one too far)

File: makejdk-any-platform-docker.sh [hunk 3]
```diff
@@ -111,9 +111,6 @@ buildOpenJDKViaDocker()
      buildDockerContainer
   fi
 
-  # Show the user all of the config before we build
-  displayParams
```
Comment (karianna) [inline makejdk-any-platform-docker.sh]: Is this run inside the container now?

File: makejdk-any-platform-docker.sh [hunk 4]
```diff
@@ -146,10 +147,38 @@ buildOpenJDKViaDocker()
...
+
+# i.e. Where we are
+SCRIPT_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+# Pull in configuration and build support
+# shellcheck source=sbin/common/config_init.sh
+source "${SCRIPT_DIR}/sbin/common/config_init.sh"
+
+# shellcheck source=native-build.sh
+source "${SCRIPT_DIR}/native-build.sh"
```
Comment (karianna) [inline makejdk-any-platform-docker.sh]: Is this needed in that case?

File: makejdk-any-platform-docker.sh [hunk 5]
```diff
@@ -146,10 +147,38 @@ buildOpenJDKViaDocker()
...
+
+# shellcheck source=configureBuild.sh
+source "${SCRIPT_DIR}/configureBuild.sh"
+
+echo "Starting $0 to configure, build (Adopt)OpenJDK binary"
+
+# Configure the build, display the parameters and write the config to disk
```
Comment (karianna) [inline makejdk-any-platform-docker.sh]: Comment is wrong as we don't display parameters?

**Comment by karianna** [comment]
> @karianna currently we configure to build on a host computer, then copy config file inside docker and use it there. It ends up with a misconfigured build which fails.
> 
> The main reason for this change is to have consistent build. So idea was to extract docker part to separate script. The build should behave the same, running native or in docker.
> 
> Does it makes more sense?

OK makes sense, I've added lots of detailed feedback on the PR.

**Comment by karianna** [comment]
> Can one of the admins verify this patch?

Still waiting for @grzesuav to update :-)

File: docker/jdk10/x86_64/ubuntu/Dockerfile [hunk 6]
```diff
@@ -55,7 +55,7 @@ RUN apt-get update \
 RUN mkdir -p /openjdk/target
```
[the remark this answers is not in the document]
Comment (grzesuav) [reply docker/jdk10/x86_64/ubuntu/Dockerfile]: removed

File: docker/jdk9/x86_64/ubuntu/Dockerfile [hunk 7]
```diff
@@ -50,7 +50,7 @@ RUN apt-get update \
 RUN mkdir -p /openjdk/target
```
[the remark this answers is not in the document]
Comment (grzesuav) [reply docker/jdk9/x86_64/ubuntu/Dockerfile]: removed

File: configureBuild.sh [hunk 8]
```diff
@@ -84,7 +84,7 @@ setWorkingDirectory()
   if [ -z "${BUILD_CONFIG[WORKSPACE_DIR]}" ] ; then
     if [[ "${BUILD_CONFIG[USE_DOCKER]}" == "true" ]];
     then
-       BUILD_CONFIG[WORKSPACE_DIR]="/openjdk/";
+       BUILD_CONFIG[WORKSPACE_DIR]="/openjdk";
```
[the remark this answers is not in the document]
Comment (grzesuav) [reply configureBuild.sh]: removed

**Comment by karianna** [comment]
@grzesuav - Is this ready for the next review?

**Comment by grzesuav** [comment]
@karianna not yet - after  merge with upstream I struggle with few things. Hope to finish it soon though.

**Comment by grzesuav** [comment]
To speed it up a bit extracted removal of unsupported jdk's to separate PR : https://github.com/AdoptOpenJDK/openjdk-build/pull/1002

**Comment by karianna** [comment]
@grzesuav any further progress here? Can we help?

**Comment by grzesuav** [comment]
I paused the work because of joining pipeline management (via gradle) in the same scripts - basically it would not work, as gradle need at least jdk8 to work (when building jdk8 we use zulu-7) and later on I read on slack that actually there is work planned to separate those scripts.

Need to check if this separation ended, if yes I can rebase and reapply the work (probably branches diverge)

**Comment by grzesuav** [comment]
https://github.com/AdoptOpenJDK/openjdk-build/issues/1108 this is the issue I have in mind.

**Comment by karianna** [comment]
> #1108 this is the issue I have in mind.

So we need to get that #1108 to be resolved first?

**Comment by grzesuav** [comment]
@karianna sorry for late response, was on holidays and conference.

Yes, currently it is impossible to build JDK8 on docker due to inclusion of gradle (used for pipelines generation I guess?) in build scripts

**Comment by adamfarley** [comment]
It seems progress on this and #1108 stalled over a year back. Should this PR and that issue be closed?

**Comment by karianna** [comment]
@grzesuav Can you rebase or close this PR?

**Comment by grzesuav** [comment]
hi @karianna , will close it as ti diverges too much

**Pull request closed without merging**